src/elements-experimental/seat-reservation/seat-reservation/seat-reservation-base-element.ts

```diff
@@ -261,42 +263,24 @@ export class SeatReservationBaseElement extends LitElement {
 
       // Set maximum calculated coach width
       this.maxCalcCoachsWidth = currCalcTriggerPos;
+    }
+  }
 
-      // At the end of a scroll Events to a coach, the reached wagon is marked as selected
-      this.coachScrollArea.addEventListener('scrollend', () => {
-        const findScrollCoachIndex = this.isAutoScrolling
-          ? this.currSelectedCoachIndex
-          : this._getCoachIndexByScrollTriggerPosition();
-
-        // In case the user uses the scrollbar without interacting with the seat reservation,
-        // the currently selected index is -1 and we have to set this value with findScrollCoachIndex.
-        if (this.currSelectedCoachIndex === -1) {
-          this.currSelectedCoachIndex = findScrollCoachIndex;
-        }
-
-        if (this._isScrollableToSelectedCoach()) {
-          this.currSelectedCoachIndex = findScrollCoachIndex;
-        } else {
-          this.currSelectedCoachIndex =
-            findScrollCoachIndex < this.currSelectedCoachIndex
-              ? this.currSelectedCoachIndex
-              : findScrollCoachIndex;
-        }
-
-        if (!this.isAutoScrolling) {
-          //When user is scrolling via scrollbar, it automatically scrolls to the focused coach in the main navigation
-          this._scrollToSelectedNavigationButton(findScrollCoachIndex);
-        }
-
-        this.preventCoachScrollByPlaceClick = false;
-        this.updateCurrentSelectedCoach();
-
-        if (!this.hasNavigation) {
-          this.preselectPlaceInCoach();
-          this.isAutoScrolling = false;
-        }
-      });
+  /**
+   * Scroll event handler managed the end of scrolling inside the coach scroll area.
+   * Timeout event handling to check if the scrolling has been completed.
+   * It is required because the Safari browser does not handle scrollend event
+   * and we therefore imitate this event -> scrollend.
+   */
+  @eventOptions({ passive: true })
+  protected coachAreaScrollend(): void {
+    if (this._scrollTimeout) {
+      // First, we cleared the registered timeout, as the scrolling is still running
+      // and thus we prevent the execution of the _handleCoachAreaScrollEvent function
+      clearTimeout(this._scrollTimeout);
     }
+    // If no further scoll event is fired, the next timeout can execute the inner function _handleCoachAreaScrollEvent without clearing
+    this._scrollTimeout = window.setTimeout(() => this._handleCoachAreaScrollEvent(), 150);
```

Comment:
doesn't matter but not needed
```suggestion
    this._scrollTimeout = setTimeout(() => this._handleCoachAreaScrollEvent(), 150);
```

Reply:
I use your suggestion, but I also had to change the return type of _scrollTimeout type -> :number to :ReturnType<typeof setTimeout>